lib/Modules/FreeRdpHooks/FreeRdpHooks.psm1

```diff
@@ -24,33 +24,26 @@ Import-Module JujuHooks
 Import-Module OpenStackCommon
 
 function Get-Vcredist {
-    BEGIN {
+    Write-JujuWarning "Getting vcredist."
+    $vcredistUrl = Get-JujuCharmConfig -Scope "vcredist-url"
+    if(!$vcredistUrl) {
         $resourcesSupport = (Get-Command resource-get.exe -ErrorAction SilentlyContinue) -ne $null
```

Comment:
Because it does not look simple, or clean :).

We have a bunch of purpose built commandlets that deal with various juju functionality. Lets use them.

It is your choice to support older versions of juju, but:

```powershell
$resourcesSupport = Get-command resource-get.exe -ErrorAction SilentlyContinue
if($resourcesSupport){
    $resource = Get-JujuResource -name vcredis
} else {
    $resource = Invoke-WebRequest -Url $vcredist
}
```

is in no way cleaner then:

```powershell
try {
   $resource = Get-JujuResource -name vcredis
} catch [NotFound] {
   # legacy juju
   $resource = Invoke-WebRequest -Url $vcredist
}